kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluator.java

```diff
@@ -566,6 +593,33 @@ ColumnVector visitCoalesce(ScalarExpression coalesce) {
             );
         }
 
+        @Override
+        ColumnVector visitTimeAdd(ScalarExpression timeAdd) {
+            ColumnVector timestampColumn = visit(timeAdd.getChildren().get(0));
+            ColumnVector durationVector = visit(timeAdd.getChildren().get(1));
+
+            long[] updatedValues = new long[timestampColumn.getSize()];
+            boolean[] nullFlags = new boolean[timestampColumn.getSize()];
+            boolean hasNulls = false;
+
+            for (int rowId = 0; rowId < timestampColumn.getSize(); rowId++) {
+                if (!timestampColumn.isNullAt(rowId) && !durationVector.isNullAt(rowId)) {
+                    long durationMicros = durationVector.getLong(rowId) * 1000L;
+                    updatedValues[rowId] = timestampColumn.getLong(rowId) + durationMicros;
+                    nullFlags[rowId] = false;
+                } else {
+                    nullFlags[rowId] = true;
+                    hasNulls = true;
+                }
+            }
+            return new DefaultLongVector(
+                TimestampType.TIMESTAMP,
+                timestampColumn.getSize(),
+                hasNulls ? Optional.of(nullFlags) : Optional.empty(),
+                updatedValues
+            );
```

Comment:
you can actually create a lazy eval ColumnVector. See the examples in this file or in LikeExpressionEvaluator